File: Data Structures & Algorithms/partition-equal-subset-sum/submission-1.java
class Solution {
    public boolean canPartition(int[] nums) {
        Arrays.sort(nums);
        int sum = 0;
        for(int num : nums) sum+=num;
        if (sum%2 == 1) return false;
        return dfs(nums, sum/2, 0);
    }

    private boolean dfs(int[] nums, int target, int index) {
        if (target == 0) return true;
        if (target < 0) return false;
        boolean b = false;
        for (int i=index; i<nums.length; i++) {
            b = dfs(nums, target-nums[i], i+1);
            if (b) break;
        }
        return b;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        arr = [0 for _ in range(10001)]

        for num in nums:
            if arr[num] > 0:
                return num
            arr[num] += 1


        # hashSet = set()
        # for num in nums:
        #     if num in hashSet:
        #         return num
        #     hashSet.add(num)

         